superset/key_value/commands/cleanup.py

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from datetime import datetime, timedelta
+
+from superset.commands.base import BaseCommand
+from superset.models.key_value import KeyValue
+from superset.key_value.utils import is_expired
+from superset.utils.celery import session_scope
+
+logger = logging.getLogger(__name__)
+
+
+class CleanupCommand(BaseCommand):
+    """
+    Expiration cleanup command.
+    """
+
+    def __init__(self, worker_context: bool = True):
+        self._worker_context = worker_context
+
+    def run(self) -> None:
+        logger.info("Key value store cleanup starting")
+        with session_scope(nullpool=True) as session:
+           for keyValue in session.query(KeyValue).all():
+                if is_expired(keyValue):
```

Comment:
It would be much better for performance to implement the logic of `is_expired` in SQL instead of Python, specially because we already have a query on line 39.

Reply:
Yes. In fact, I'm currently doing this exact change. I'm just trying to learn how to do that with SQL Alchemy since this is my first Python PR :sweat_smile:

Reply:
Oh, good job then! I could never tell this was your first Python PR!

And yeah, doing SQL with SQLAlchemy is not fun... :-/

Reply:
There must be some way to restrict this full table scan `session.query(KeyValue).all()`. What if, when we create/update a key we timestamp an expiration date time then filter by it?

Reply:
I'm on it. A DELETE with a WHERE clause and a simple condition will be sufficient.